### DISCUSSION TITLE
Trouble with invite E-Mails

### DISCUSSION THREAD

**Discussion by xgrnsxs** [TandoorRecipes/recipes#1515] [General]
Hi
I really like this project, so i set up a family wide Recipe Database. When I create an invite link without an e-mail adress, it works fine. But if i enter any E-Mail Adress I get an HTTP ERROR 500 immediately.

I don't see anything E-Mail related popping up neither in the gunicorn_error.log nor in journal or nginx logs. I tried setting up a test-mailserver  and i don't even see tandoor trying to connect to it.

Can you give me some hints where to look?
If you need some specific info about my setup, please ask, but i followed the manual setup guide pretty much 100%.
thanks very much!

**Comment by xgrnsxs** [comment]
I just stumbled upon the following command and it gives me an error as follows:

root@tandoor:/var/www/recipes# python manage.py sendtestemail mail@domain.com
  File "manage.py", line 14
    ) from exc
         ^
SyntaxError: invalid syntax

**Comment by smilerz** [comment]
how is your environment setup?

**Comment by xgrnsxs** [comment]
LOL...
I just found out, that I missed the step, to install all the modules from the requirements.txt, so now i can use "sendtestemail" without error, but I still get Server ERROR 500 when I create invite links with an E-Mail Address.

Okay, so my Environment is as follows:

- LXC Container with Ubuntu 20.04
- PSQL 13.5
- Django 3.2.11
- nodejs 12.22.5
- python 3.9.2
- nginx as reverse proxy to /var/www/recipes/recipes.sock;

Attached the .env file (no passwords or secrets of course :D )
[env.txt](https://github.com/TandoorRecipes/recipes/files/8027357/env.txt)

If you need something specific, please let me know, as I am not very familiar with python/django.

**Reply by smilerz** [reply]
and your nginx config

**Reply by smilerz** [reply]
Can you set DEBUG=1 and share the django logs showing the error?

**Reply by xgrnsxs** [reply]
Sure and by asking this, i found the error!

In the .env the line with the option "DEFAULT_FROM_EMAIL" my Mail Adress was follwed by a # and a comment on the same line. It might have been there all the time, or maybe i messed it up. Because I did not know that comments cannot be in the same line with any options, I did not see the error.

Thank you very much, it works!

For the sake of completeness here are the logs:

[2022-02-08 22:01:56 +0100] [14549] [DEBUG] POST /new/invite-link/
/var/www/recipes/lib/python3.9/site-packages/django/db/models/fields/__init__.py:1416: RuntimeWarning: DateTimeField InviteLink.created_at received a naive datetime (2022-02-08 18:01:56.497966) while time zone support is active.
  warnings.warn("DateTimeField %s received a naive datetime (%s)"
Internal Server Error: /new/invite-link/
Traceback (most recent call last):
  File "/var/www/recipes/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/var/www/recipes/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/var/www/recipes/lib/python3.9/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/var/www/recipes/cookbook/helper/permission_helper.py", line 151, in dispatch
    return super(GroupRequiredMixin, self).dispatch(request, *args, **kwargs)
  File "/var/www/recipes/lib/python3.9/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/var/www/recipes/lib/python3.9/site-packages/django/views/generic/edit.py", line 172, in post
    return super().post(request, *args, **kwargs)
  File "/var/www/recipes/lib/python3.9/site-packages/django/views/generic/edit.py", line 142, in post
    return self.form_valid(form)
  File "/var/www/recipes/cookbook/views/new.py", line 219, in form_valid
    send_mail(
  File "/var/www/recipes/lib/python3.9/site-packages/django/core/mail/__init__.py", line 61, in send_mail
    return mail.send()
  File "/var/www/recipes/lib/python3.9/site-packages/django/core/mail/message.py", line 284, in send
    return self.get_connection(fail_silently).send_messages([self])
  File "/var/www/recipes/lib/python3.9/site-packages/django/core/mail/backends/smtp.py", line 109, in send_messages
    sent = self._send(message)
  File "/var/www/recipes/lib/python3.9/site-packages/django/core/mail/backends/smtp.py", line 121, in _send
    from_email = sanitize_address(email_message.from_email, encoding)
  File "/var/www/recipes/lib/python3.9/site-packages/django/core/mail/message.py", line 88, in sanitize_address
    raise ValueError(
ValueError: Invalid address; only {mail-address censored}#  could be parsed from "{mail-address-censored}# email sender address (default 'webmaster@localhost')"

**Comment by xgrnsxs** [comment]
Ah, sry, here you go:
I have both HTTP and HTTPS set up:

For HTTP:
server {
    listen 8002;
    #access_log /var/log/nginx/access.log;
    #error_log /var/log/nginx/error.log;

    # serve media files
    location /static {
        alias /var/www/recipes/staticfiles;
    }

    location /media {
        alias /var/www/recipes/mediafiles;
    }

    location / {
        proxy_set_header Host $http_host;
        proxy_pass http://unix:/var/www/recipes/recipes.sock;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}

For HTTPS:
server {
    listen 443 http2 ssl;
    listen [::]:443 http2 ssl;

    server_name {mydomain};

    ssl_certificate /etc/ssl/certs/rezepte.crt;
    ssl_certificate_key /etc/ssl/private/rezepte.key;
    ssl_dhparam /etc/ssl/certs/dhparam.pem;

    location /static {
        alias /var/www/recipes/staticfiles;
    }

    location /media {
        alias /var/www/recipes/mediafiles;
    }

    location / {
        proxy_set_header Host $http_host;
        proxy_pass http://unix:/var/www/recipes/recipes.sock;
        proxy_set_header X-Forwarded-Proto $scheme;

    }
}
